src/Framework/QueryBuilder/Concerns/MetaQuery.php

```diff
@@ -108,7 +97,17 @@ public function attachMeta($table, $foreignKey, $primaryKey, ...$columns)
             // Set dynamic alias
             $tableAlias = sprintf('%s_%s_%d', ($table instanceof RawSQL) ? $table->sql : $table, 'attach_meta', $i);
 
+            // Check if we have meta columns that dev wants to group concat
             if ($concat) {
+                /**
+                 * Include foreign key to prevent errors if sql_mode is only_full_group_by
+                 *
+                 * @see https://dev.mysql.com/doc/refman/5.7/en/group-by-handling.html
+                 */
+                $this->groupBy($foreignKey);
+
+                // Group concat same key values into faux array
+                // @example key => ["value1", "value2"]
                 $this->selectRaw(
                     "CONCAT('[',GROUP_CONCAT(DISTINCT CONCAT('\"',%1s,'\"')),']') AS %2s",
                     $tableAlias . '.' . $metaTable->valueColumnName,
```

Comment:
@jonwaldstein What's the scenario for this? Is this for handling meta that has the same key across multiple rows? Have a tested this?

Reply:
Have you done an `EXPLAIN` on this? Are we still avoiding a temporary table?

Reply:
Yes, the `EXPLAIN` query I reported above _includes_ `additional_email`.